src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```
@@ -0,0 +1,472 @@
+                        list.Add(t);
+                        return Tuple.Create(TheAction.Add, list.Count - 1, -1);
+                    }
+                })
+                .ObserveOn(scheduler)
+                .Select(x =>
+                {
+                    UpdateFilter(x.Item1, x.Item2, x.Item3);
+                    return Unit.Default;
+                })
+                .Publish()
+                .RefCount();
+            return this;
+        }
+
+        public void SetComparer(Func<T, T, int> compare)
+        {
+            comparer = compare;
+            var list = filter == null ? this : original as IList<T>;
+            UpdateSort(list, 0, list.Count);
+            UpdateFilter(filter);
+        }
+
+        public void SetFilter(Func<T, int, IList<T>, bool> theFilter)
+        {
+            UpdateFilter(theFilter);
+        }
+
+        void UpdateSort(IList<T> list, int start, int end)
+        {
+            if (comparer == null)
+                return;
+
+            for (; start < end - 1; start++)
+            {
+                var idxSmallest = start;
+                for (var i = start + 1; i < end; i++)
+                    if (comparer(list[i], list[idxSmallest]) < 0)
+                        idxSmallest = i;
+                if (idxSmallest == start) continue;
+                Swap(list, start, idxSmallest);
+
+                // if there's a filter, then it's going to trigger events and we don't need to manually trigger them
+                if (filter == null)
+                {
+                    OnPropertyChanged(new PropertyChangedEventArgs("Item[]"));
+                    OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Move, this[start], start, idxSmallest));
+                }
+            }
+        }
+
+        void UpdateFilter(Func<T, int, IList<T>, bool> newFilter)
+        {
+            if (filter == null && newFilter == null)
+                return; // nothing to do
```

Is this even valid to do this? Should we throw an exception to indicate that this is a mistake? Also, what about the case where `newFilter == filter`? An edge case, but we should probably just handle it too.